pymc/variational/opvi.py

```diff
@@ -956,42 +976,45 @@ def __init_group__(self, group):
         self.input = self._input_type(self.__class__.__name__ + "_symbolic_input")
         # I do some staff that is not supported by standard __init__
         # so I have to to it by myself
-        self.group = [get_transformed(var) for var in self.group]
 
-        # XXX: This needs to be refactored
-        # self.ordering = ArrayOrdering([])
-        self.replacements = dict()
+        # 1) we need initial point (transformed space)
+        model_initial_point = self.model.compute_initial_point(0)
+        # 2) we'll work with a single group, a subset of the model
+        # here we need to create a mapping to replace value_vars with slices from the approximation
+        start_idx = 0
         for var in self.group:
             if var.type.numpy_dtype.name in discrete_types:
                 raise ParametrizationError(f"Discrete variables are not supported by VI: {var}")
-            begin = self.ddim
+            # 3) This is the way to infer shape and dtype of the variable
+            value_var = self.model.rvs_to_values[var]
+            test_var = model_initial_point[value_var.name]
             if self.batched:
+                # Leave a more complicated case for future work
                 if var.ndim < 1:
                     if self.local:
                         raise LocalGroupError("Local variable should not be scalar")
                     else:
                         raise BatchedGroupError("Batched variable should not be scalar")
-                # XXX: This needs to be refactored
-                # self.ordering.size += None  # (np.prod(var.dshape[1:])).astype(int)
+                size = test_var[0].size
                 if self.local:
-                    # XXX: This needs to be refactored
-                    shape = None  # (-1,) + var.dshape[1:]
+                    shape = (-1,) + test_var.shape[1:]
                 else:
-                    # XXX: This needs to be refactored
-                    shape = None  # var.dshape
+                    shape = test_var.shape
             else:
-                # XXX: This needs to be refactored
-                # self.ordering.size += None  # var.dsize
-                # XXX: This needs to be refactored
-                shape = None  # var.dshape
-            # end = self.ordering.size
-            # XXX: This needs to be refactored
-            vmap = None  # VarMap(var.name, slice(begin, end), shape, var.dtype)
-            # self.ordering.vmap.append(vmap)
-            # self.ordering.by_name[vmap.var] = vmap
-            vr = self.input[..., vmap.slc].reshape(shape).astype(vmap.dtyp)
-            vr.name = vmap.var + "_vi_replacement"
-            self.replacements[var] = vr
+                shape = test_var.shape
+                size = test_var.size
+            # TODO: There was self.ordering used in other util funcitons
```

Comment:
What about this TODO?

Reply:
don't really remember, I'd rather remove that at all

Reply:
```suggestion
```